cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -58,6 +60,25 @@ public static void deleteGCSPath(Storage storageClient, String bucket, String pa
     }
   }
 
+  /**
+   * Utilty class to parse the keyvalue string from UI Widget and return back HashMap
+   * @param configValue
+   * @param delimiter
+   * @param keyValueDelimiter
+   * @return
+   */
+  public static Map<String, String> parseKeyValueConfig(String configValue, String delimiter,
+                                                        String keyValueDelimiter) {
+    Map<String, String> map = new HashMap<>();
+    for (String property : configValue.split(delimiter)) {
+      String[] parts = property.split(keyValueDelimiter);
+      String key = parts[0];
```

Comment:
Validate the array size is 2 before using it.

Reply:
Done